post_processing/ci_plots.jl

```diff
@@ -590,22 +623,112 @@ EDMFBoxPlots = Union{
     Val{:prognostic_edmfx_trmm_column},
 }
 
+"""
+    plot_edmf_vert_profile!(grid_loc, var_group)
+
+Helper function for `make_plots_generic`. Takes a list of variables and plots
+them on the same axis.
+"""
+function plot_edmf_vert_profile!(grid_loc, var_group)
+    z = var_group[1].dims["z"]
+    units = var_group[1].attributes["units"]
+    ax = CairoMakie.Axis(
+        grid_loc[1, 1],
+        ylabel = "z [$(var_group[1].dim_attributes["z"]["units"])]",
+        xlabel = "$(short_name(var_group[1])) [$units]",
+        title = var_info(var_group[1]),
+    )
+
+    for var in var_group
+        CairoMakie.lines!(ax, var.data, z, label = short_name(var))
+    end
+    length(var_group) > 1 && Makie.axislegend(ax)
+end
+
+
+"""
+    plot_custom_title!(grid_loc, var)
+
+Helper function for `make_plots_generic`. Plots a variable with a custom title.
+"""
+plot_custom_title!(grid_loc, var) = viz.plot!(
+    grid_loc,
+    var;
+    more_kwargs = Dict(:axis => ca_kwargs(title = var_info(var))),
+)
+
+"""
+    group_edmf_vars(vars)
+
+Groups updraft and gridmean EDMF OutputVars into a Dict.
+Groupings are matched via variable short name, location, and time (if applicable).
+"""
+function group_edmf_vars(vars)
+    grouped_vars = OrderedDict{String, Vector{ClimaAnalysis.OutputVar}}()
+    for var in sort(vars, by = var_info)
+        suffix = var_info(var)
+        if haskey(grouped_vars, suffix)
+            push!(grouped_vars[suffix], var)
+        else
+            grouped_vars[suffix] = [var]
+        end
+    end
+    return grouped_vars
+end
```

Comment:
I find this function hard to understand.

Why does the dictionary have to be ordered?
Why do the variables have to be sorted, and why sorted by `var_info`?

What I understand is that you are matching some variables with their `up` version. The trouble comes from the fact that there's also slicing information. What if you do the matching before slicing? See below